### PR TITLE
Allowed multiple URLs in background-image property #1747

### PR DESCRIPTION
**Pull request by mj3cheun**
Did basic testing and did not notice anything odd. If the top image has transparency, one can see the image underneath. They are basically all stacked on each other. Also designed changes to be resistant to race-condition bugs, however, cannot be sure none exist as they are hard to test.

### CODE REVIEW

**Comment by maxkfranz** [comment]
I'm getting errors on the debug page and it seems that the `background-image` property isn't getting the proper result for its parsed object representation.

I made commit ad1337dc23bab7013882ac2a7c68b785d6a2a97e to give you an example of how the property definition should work.  If you need multiple support on a type but the type could potentially be used singly elsewhere, then a plural type with a similar spec should be defined.  See `url` and `urls` as an example.  (It should also simplify your `url[][]` code to `url[]`.)

You'll have to `git rebase ad1337dc23bab7013882ac2a7c68b785d6a2a97e` or merge to put the change in your fork.

Most of the background image properties also need to have multiple variants or be made multiple themselves:

```js
    // node background images
    { name: 'background-image', type: t.url }, // urls
    { name: 'background-image-crossorigin', type: t.bgCrossOrigin }, // chg to mult
    { name: 'background-image-opacity', type: t.zeroOneNumber }, // zeroOneNumbers
    { name: 'background-position-x', type: t.bgPos }, // chg to mult
    { name: 'background-position-y', type: t.bgPos },// chg to mult
    { name: 'background-repeat', type: t.bgRepeat }, // chg to mult
    { name: 'background-fit', type: t.bgFit }, // chg to mult
    { name: 'background-clip', type: t.bgClip }, // this makes sense as once per node
    { name: 'background-width', type: t.bgWH }, // chg to mult
    { name: 'background-height', type: t.bgWH }, // chg to mult
```

Basically, each image needs to be defined independently for the most part.

Does that make sense?

**Comment by mj3cheun** [comment]
Ok I am done making my fixes. Note that the tests fail because getting one numeric value for a property with potentially multiple values does not make any sense. Those tests may have to be modified.

File: src/collection/style.js [hunk 1]
```diff
@@ -71,6 +71,25 @@ var elesfn = ({
     }
   },
 
+  // get the associated values at the specified index when parsing a property with a multiple tag
+  parsedStyleMult: function( property, index ){
+    var ele = this[0];
```
Comment (maxkfranz) [inline src/collection/style.js]: - This is too expensive to run for reading properties.
- It's potentially mutating the style for the element.
- It removes other needed info by only giving you the `value`.  You often need the `pfValue` and/or the `units` as well.

You can't have a general function like this for style properties.  

You should define a general `util.arrayValueAt( array, index, fallbackIndex )` function in `util` and use `util.arrayValueAt( ele.pstyle('foo').value, i, 0 )`.

File: src/extensions/renderer/canvas/drawing-nodes.js [hunk 2]
```diff
@@ -155,12 +152,21 @@ CRp.drawNode = function( context, node, shiftToOriginWithBb, drawLabel ){
...
-  if( urlDefined ){
-    if( image.complete ){
-      this.drawInscribedImage( context, image, node );
+  var prevBging = _p.backgrounding;
+  var totalCompleted = 0;
+
+  for( var i = 0; i < numImages; i++ ){
+    if( ( urlDefined[i] ) && image[i].complete ){
+      totalCompleted++;
+      this.drawInscribedImage( context, image, node, i );
```
Comment (maxkfranz) [inline src/extensions/renderer/canvas/drawing-nodes.js]: You should be passing in `image[i]`.  The function doesn't need an array of images.

File: src/collection/style.js (hunk 1 above)
Comment (mj3cheun) [inline src/collection/style.js]: Ok, so at the moment, if the index is not found, it defaults to the default settings in getDefaultProperty. So it is ok to default to index 0 instead?

Comment (maxkfranz) [inline src/collection/style.js]: If you want to use the default you need to ` || defaultPropertyValue` for each property as it's read (`pfValue`, `value`, etc.).  You can't use the approach you have above, especially because you can't ever modify `style`.  Only the styler does that, and nothing changes the properties once they're created.
